src/tzdb.rs

```diff
@@ -364,35 +364,73 @@ fn resolve_posix_tz_string_for_epoch_seconds(
 
     let transition =
         compute_tz_for_epoch_seconds(is_transition_day, transition, seconds, dst_variant);
-    let offset = match transition {
-        TransitionType::Dst => {
-            LocalTimeRecord::from_daylight_savings_time(&dst_variant.variant_info)
-        }
-        TransitionType::Std => LocalTimeRecord::from_standard_time(&posix_tz_string.std_info),
+    let std_offset = LocalTimeRecord::from_standard_time(&posix_tz_string.std_info).offset;
+    let dst_offset = LocalTimeRecord::from_daylight_savings_time(&dst_variant.variant_info).offset;
+    let (old_offset, new_offset) = match transition {
+        TransitionType::Dst => (std_offset, dst_offset),
+        TransitionType::Std => (dst_offset, std_offset),
     };
     let transition = match transition {
         TransitionType::Dst => start,
         TransitionType::Std => end,
     };
-    let year = utils::epoch_time_to_epoch_year(seconds);
+    let year = utils::epoch_time_to_epoch_year(seconds * 1000);
     let year_epoch = utils::epoch_days_for_year(year) * 86400;
-    let leap_days = utils::mathematical_days_in_year(year) - 365;
+    let leap_day = utils::mathematical_in_leap_year(seconds * 1000) as u16;
 
     let days = match transition.day {
-        TransitionDay::NoLeap(day) if day > 59 => i32::from(day) - 1 + leap_days,
-        TransitionDay::NoLeap(day) => i32::from(day) - 1,
-        TransitionDay::WithLeap(day) => i32::from(day),
-        TransitionDay::Mwd(_month, _week, _day) => {
-            // TODO: build transition epoch from month, week and day.
-            return Ok(TimeZoneOffset {
-                offset: offset.offset,
-                transition_epoch: None,
-            });
+        TransitionDay::NoLeap(day) if day > 59 => day - 1 + leap_day,
+        TransitionDay::NoLeap(day) => day - 1,
+        TransitionDay::WithLeap(day) => day,
+        TransitionDay::Mwd(month, week, day) => {
+            let days_to_month = utils::month_to_day((month - 1) as u8, leap_day);
```

Comment:
nit: it may be a bit more clear for this to be renamed from `month_to_day` -> `month_to_days_of_year`.

I think I had originally written this function, and looking back I think the name lacks a bit of clarity.

Reply:
I'm planning on addressing this in a future cleanup that makes the names on the `utils` module a bit more consistent.